Udiscovery v1.5.0

jvnaiduwin opened the pull request:
Eclipse uProtocol 1.5 Refactor

Integrated new uprotocol-java SDK
Refactored unit tests
Remove synchronization with Central Discovery Service
  - factory reset initialization
  - intersection logic
  - remote RPC calls
Remove uRegistry Database
Remove hardcoded URI strings
Remove Authenticator
Remote Tag, Key and Log

Commit:
Initial implementation of uProtocol core services

Commit:
Eclipse uProtocol 1.5 Refactor

Integrated new uprotocol-java SDK
Refactored unit tests
Remove synchronization with Central Discovery Service
  - factory reset initialization
  - intersection logic
  - remote RPC calls
Remove uRegistry Database
Remove hardcoded URI strings
Remove Authenticator
Remote Tag, Key and Log

Commit:
Eclipse uProtocol 1.5 Refactor

Integrated new uprotocol-java SDK
Refactored unit tests
Remove synchronization with Central Discovery Service
  - factory reset initialization
  - intersection logic
  - remote RPC calls
Remove uRegistry Database
Remove hardcoded URI strings
Remove Authenticator
Remote Tag, Key and Log

Commit:
Eclipse uProtocol 1.5 Refactor

Integrated new uprotocol-java SDK
Refactored unit tests
Remove synchronization with Central Discovery Service

    factory reset initialization
    intersection logic
    remote RPC calls
    Remove uRegistry Database
    Remove hardcoded URI strings
    Remove Authenticator
    Remote Tag, Key and Log
    EnableExpiryTable and Unit test

mishap4 commented on up-discovery-android/Android.bp:
```diff
@@ -0,0 +1,27 @@
+gradle_app {
```
I believe we shouldn't provide soong build script for this project. Please remove Android.bp

jvnaiduwin replied:
Done.

mishap4 commented on up-discovery-android/NOTICE:
```diff
@@ -0,0 +1,3866 @@
+Copyright 2023
```
I don't think we need to include GM NOTICE file.

jvnaiduwin replied:
NOTICE is included in the license comment

mishap4 commented on up-discovery-android/README.adoc:
```diff
@@ -0,0 +1,27 @@
+----
+
+=== Installing the Build
+Please follow xref:../testing.adoc#installing-a-privileged-application[instructions] to install _UDiscoveryService.apk_ as a privileged application.
```
Please rebase, there was an update on v1.5.0 branch to replace "xref" with "link"

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/gradle/jacoco.gradle:
```diff
@@ -0,0 +1,48 @@
+    sourceDirectories.from = ["${projectDir}/src/main/java"]
+    classDirectories.from = fileTree(
+            dir: "${buildDir}/intermediates/javac/debug/classes/",
+            excludes: ['**/IU*', '**/BuildConfig.class', '**/*_Impl*.*']
```
Do you have any classes matching '**/IU*' or '**/*_Impl*.*' in this project?

jvnaiduwin replied:
removed all, included impl

Commit:
UDiscovery UT

-UDiscovery Unit Test update
-Refactor variable names
-Android entry for entity name and version

Commit:
Rename variable

Rename the package name

mishap4 commented on up-discovery-android/privapp-permissions-org.eclipse.uprotocol.core.udiscovery.xml:
```diff
@@ -0,0 +1,32 @@
+<permissions>
+    <privapp-permissions package="org.eclipse.uprotocol.core.udiscovery">
+        <permission name="android.permission.INTERACT_ACROSS_USERS"/>
+        <permission name="android.permission.DUMP" />
```
Are you using any APIs which require DUMP, PACKAGE_USAGE_STATS or CAR_VENDOR_EXTENSION?

jvnaiduwin replied:
removed permission.

mishap4 commented on up-discovery-android/service/build.gradle:
```diff
@@ -0,0 +1,118 @@
+        release {
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+            signingConfig signingConfigs.platform
+            minifyEnabled false
```
Why not to enable for release build?

jvnaiduwin replied:
Please clarify.

mishap4 commented on up-discovery-android/service/src/main/AndroidManifest.xml:
```diff
@@ -0,0 +1,72 @@
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <uses-permission android:name="ultifi.permission.ACCESS_UBUS" />
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
```
I believe you don't need all permissions declared here. Please leave only those which are required

jvnaiduwin replied:
As per understanding most of unwanted permission removed.

mishap4 commented on up-discovery-android/service/src/main/AndroidManifest.xml:
```diff
@@ -0,0 +1,72 @@
+                android:name="uprotocol.entity.version"
+                android:value="3" />
+            <meta-data
+                android:name="uprotocol.startup.trigger"
```
There is no uServiceController which starts services. This config is not needed. Need to think how we are going to start service

jvnaiduwin replied:
Removed startup trigger

mishap4 commented on up-discovery-android/service/src/main/AndroidManifest.xml:
```diff
@@ -0,0 +1,72 @@
+                android:name="uprotocol.startup.user"
+                android:value="system" />
+            <meta-data
+                android:name="uprotocol.entity.permissionLevel"
```
No CAP feature available. It is also not needed

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/assets/udiscovery/udiscovery.json:
```diff
@@ -0,0 +1,128 @@
+  "uri": "ultifi:/core.udiscovery",
+  "nodes": [
+    {
+      "uri": "ultifi:/core.udiscovery/2",
```
It is version 3, right?

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/gradle/libs.versions.toml:
```diff
@@ -0,0 +1,60 @@
+robolectric = "4.7.3"
+room = "2.4.2"
+room-testing = "1.1.1"
+uprotocol-java = "1.5.6-SNAPSHOT"
```
1.5.6 has been already released

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/assets/udiscovery/udiscovery.json:
```diff
@@ -0,0 +1,128 @@
+            "id": {
+              "uInteger": 2
+            },
+            "method_permission_level": {
```
CAP is not implemented, remove all permission related props

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/doxygen/Doxyfile:
```diff
@@ -0,0 +1,2737 @@
+# Doxyfile 1.9.4
```
Why do we need doxygen?

jvnaiduwin replied:
Team recommend to have doxygen. If you still want us to delete let me know your concern.

jvnaiduwin replied:
Team recommend to have doxygen.

mishap4 commented on up-discovery-android/service/src/main/doxygen/ReadMe.md:
```diff
@@ -0,0 +1,39 @@
+# How to Install & Run Doxygen for IV UDiscoveryService
+
```
Same as above

jvnaiduwin replied:
Team recommend to have doxygen. If you still want us to delete let me know your concern.

jvnaiduwin replied:
Team recommend to have doxygen.

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/AssetUtility.java:
```diff
@@ -0,0 +1,84 @@
+
+public class AssetUtility {
+
+    private static final String LOG_TAG = Formatter.tag("core", AssetUtility.class.getSimpleName());
```
With minify enabled AssetUtility.class will be renamed. Do we want to keep original name in logs? Use string string literal then. (Here and everywhere else)

jvnaiduwin replied:
Done. Now using Service.getName()

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/AssetUtility.java:
```diff
@@ -0,0 +1,84 @@
+
+public class AssetUtility {
+
+    private static final String LOG_TAG = Formatter.tag("core", AssetUtility.class.getSimpleName());
```
Why not to use just TAG, like in other projects?  Please rename (Here and everywhere else)

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/AssetUtility.java:
```diff
@@ -0,0 +1,84 @@
+        try {
+            final String filepath = context.getFilesDir() + "/" + sFileName;
+            final StringBuilder buffer = new StringBuilder();
+            BufferedReader reader = new BufferedReader(new FileReader(filepath));
```
final can be used as well

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/AssetUtility.java:
```diff
@@ -0,0 +1,84 @@
+    public AssetUtility() {
+    }
+
+    public String readFileFromInternalStorage(Context context, String sFileName) {
```
java naming conversion for params tells "fileName" (here and everywhere else)

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/LoadUtility.java:
```diff
@@ -0,0 +1,101 @@
+        Log.d(LOG_TAG, join(Key.EVENT, "load", Key.STATUS, "successful"));
+    }
+
+    enum initLDSCode {
```
Java naming convention tells: "Init..."

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/AssetUtility.java:
```diff
@@ -0,0 +1,84 @@
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class AssetUtility {
```
AssetManager?

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/LoadUtility.java:
```diff
@@ -0,0 +1,101 @@
+import org.eclipse.uprotocol.common.util.log.Key;
+import org.eclipse.uprotocol.core.udiscovery.db.DiscoveryManager;
+
+public class LoadUtility {
```
Why not call it as DatabaseLoader or some other Loader. Utility sounds like static util class

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/LoadUtility.java:
```diff
@@ -0,0 +1,101 @@
+        return code;
+    }
+
+    private boolean saveDB(String filename) {
```
Just "save"? as you have "load"

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/Notifier.java:
```diff
@@ -0,0 +1,162 @@
+    protected static boolean DEBUG = Log.isLoggable(LOG_TAG, Log.DEBUG);
+    protected static boolean VERBOSE = Log.isLoggable(LOG_TAG, Log.VERBOSE);
+    private final ObserverManager mObserverManager;
+    private final ULink mEUlink;
```
Please just mULink (Here and everywhere else)

jvnaiduwin replied:
Change to mUpClient

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/Notifier.java:
```diff
@@ -0,0 +1,162 @@
+    private final ObserverManager mObserverManager;
+    private final ULink mEUlink;
+
+    public Notifier(ObserverManager observerManager, ULink ultifiLink) {
```
We shouldn't use "Ultify" anywhere in this project (Here and everywhere else)

jvnaiduwin replied:
Done. Commented code line not removed

mishap4 commented on up-discovery-android/settings.gradle:
```diff
@@ -0,0 +1,45 @@
+
+apply from: 'gradle/config.gradle'
+
+rootProject.name = "UDiscoveryService"
```
"uDiscovery Service" as a project name like in other projects

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/internal/log/Formatter.java:
```diff
@@ -0,0 +1,134 @@
+
+import org.eclipse.uprotocol.common.util.log.Key;
+
+public final class Formatter {
```
Formatter is available in up-client-android-library: ./library/src/main/java/org/eclipse/uprotocol/common/util/log/Formatter.java
Remove this class. If you need to extend Create FormatterExt like it is done in uCore service project

jvnaiduwin replied:
removed this class. I guess this class is introduced before library.

mishap4 commented on up-discovery-android/service/src/main/res/values/strings.xml:
```diff
@@ -0,0 +1,5 @@
+<resources>
+    <string name="app_name">UDiscoveryService</string>
+    <string name="deviceId">CSM_VIN_2022001</string>
```
What is "CSM" in open source project?

jvnaiduwin replied:
Removed, I guess this available from last project refactor

mishap4 commented on up-discovery-android/service/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Can be removed since minSdk is greater then v26

jvnaiduwin replied:
Removed

mishap4 commented on up-discovery-android/service/src/main/res/mipmap-anydpi-v26/ic_launcher.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Can be removed since minSdk is greater then v26

jvnaiduwin replied:
Removed

mishap4 commented on up-discovery-android/service/src/main/res/drawable/ic_launcher_background.xml:
```diff
@@ -0,0 +1,170 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Can be removed since used only in v26 resource

jvnaiduwin replied:
Removed

mishap4 commented on up-discovery-android/service/src/main/res/drawable-v24/ic_launcher_foreground.xml:
```diff
@@ -0,0 +1,30 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Can be removed since used only in v26 resource

jvnaiduwin replied:
Removed

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/db/ExpiryData.java:
```diff
@@ -0,0 +1,40 @@
+
+public class ExpiryData {
+    public final String uri;
+    public final String exp;
```
what is exp? Why not  give full name fut -> future?

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/db/DiscoveryManager.java:
```diff
@@ -0,0 +1,592 @@
+
+public class DiscoveryManager {
+    private static final String LOG_TAG = Formatter.tag("core", DiscoveryManager.class.getSimpleName());
+    private static final String LOCALURI = "localUri";
```
LOCAL_URI?

jvnaiduwin replied:
This just log tag or variable use to print in logs. KEY and events.

jvnaiduwin replied:
repalce with Key.URI

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/db/DiscoveryManager.java:
```diff
@@ -0,0 +1,592 @@
+    private static final String LOG_TAG = Formatter.tag("core", DiscoveryManager.class.getSimpleName());
+    private static final String LOCALURI = "localUri";
+    private static final String KEY = "key";
+    private final Gson gson = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
```
Please select one common for the project naming style for member fileds "gson" vs "mGson"

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/db/DiscoveryManager.java:
```diff
@@ -0,0 +1,592 @@
+    private final ExpiryTable expiryTable = new ExpiryTable();
+    private final Notifier mNotifier;
+    private final ExecutorService mExecutor = Executors.newSingleThreadExecutor();
+    private org.eclipse.uprotocol.v1.UAuthority ldsAuthority = UAuthority.getDefaultInstance();
```
can be imported, right?

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/db/DiscoveryManager.java:
```diff
@@ -0,0 +1,592 @@
+     * Returns: [“ultifi://ultifi.gm.com/core.example/2.0”, “ultifi:core.example/1.0”]
+     */
+    public synchronized Pair<UUriBatch, UStatus> lookupUri(@NonNull UUri uri) {
+        UStatus sts;
```
Why not use "status" as name?

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/db/DiscoveryManager.java:
```diff
@@ -0,0 +1,592 @@
+            if (serviceName.equals(JSON_AUTHORITY)) {
+                final UUri authorityUri = UUri.newBuilder().setAuthority(ldsAuthority).build();
+                batch.addUris(authorityUri);
+                sts = UStatus.newBuilder().setCode(UCode.OK).setMessage("[lookupUri] Success").build();
```
you may use buildStatus utils to make it shorter

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/db/DiscoveryManager.java:
```diff
@@ -0,0 +1,592 @@
+            final Node serviceNode = DatabaseUtility.internalFindNode(ldsTree, serviceUri);
+            checkNotNull(serviceNode, UCode.NOT_FOUND, "[lookupUri] could not find " + serviceUri);
+            for (Node entity : serviceNode.getNodesList()) {
+                final UUri fullUri = LongUriSerializer.instance().deserialize(entity.getUri());
```
You have toLongUri, maybe add fromLongUri for consistency?

jvnaiduwin replied:
Done

jvnaiduwin replied:
added method fromLongUri in Utils class

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/NwConnectionCallbacks.java:
```diff
@@ -0,0 +1,68 @@
+import org.eclipse.uprotocol.common.util.log.Key;
+import org.eclipse.uprotocol.core.udiscovery.interfaces.NetworkStatusInterface;
+
+public class NwConnectionCallbacks extends ConnectivityManager.NetworkCallback {
```
Can we rename it to NetworkCallback?

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/RPCHandler.java:
```diff
@@ -0,0 +1,346 @@
+        return status;
+    }
+
+    private UStatus UnregisterForNotifications(@NonNull UUri observer, @NonNull List<UUri> nodeUriList) {
```
Lowercase please: unregisterForNotifications

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/RPCHandler.java:
```diff
@@ -0,0 +1,346 @@
+        mAssetUtil.writeFileToInternalStorage(mContext, Constants.LDS_DB_FILENAME, data);
+    }
+
+    public UPayload processLookupUriFromLDS(@NonNull UMessage uMsg) {
```
uMsg -> message

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/RPCHandler.java:
```diff
@@ -0,0 +1,346 @@
+            } else if (UDiscoveryService.METHOD_UNREGISTER_FOR_NOTIFICATIONS.equals(methodName)) {
+                status = UnregisterForNotifications(observerUri, nodeUriList);
+            } else {
+                final UCode code = UCode.INVALID_ARGUMENT;
```
Can be inline: throw new UStatusException(UCode.INVALID_ARGUMENT,  "unknown method " + methodName);

jvnaiduwin replied:
This legacy code which shared to me. Need to check with team can we can change.

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/common/Constants.java:
```diff
@@ -0,0 +1,56 @@
+
+    public static final String UDISCOVERY = "udiscovery";
+    public static final String UDISCOVERY_JSON = "udiscovery.json";
+    public static final UAuthority LDS_AUTHORITY = UAuthority.newBuilder().setName(
```
What is this? Placeholder?

jvnaiduwin replied:
removed

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/common/Constants.java:
```diff
@@ -0,0 +1,56 @@
+    public static final UAuthority LDS_AUTHORITY = UAuthority.newBuilder().setName(
+            String.join(".", LDS_DEVICE_NAME, LDS_DOMAIN_NAME)).build();
+
+    public static final String PERMISSION_LEVEL = "permission_level";
```
Do we need it? CAP is not implemented

jvnaiduwin replied:
Removed

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/common/LogUtils.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    @SuppressLint("DefaultLocale")
+    public static String toPrettyMemory(long bytes) {
```
We have this method in Formatter from up-client-android-java library. Please remove this class

jvnaiduwin replied:
Removed

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/UDiscoveryService.java:
```diff
@@ -0,0 +1,447 @@
+@SuppressWarnings({"java:S1200", "java:S3008", "java:S1134"})
+public class UDiscoveryService extends Service implements NetworkStatusInterface {
+    public static final String LOG_TAG = Formatter.tag("core", UDiscoveryService.class.getSimpleName());
+    public static final UEntity UDISCOVERY_SERVICE = UEntity.newBuilder().setName("core.udiscovery").setVersionMajor(3).build();
```
Can be just SERVICE or ENTITY constant

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/UDiscoveryService.java:
```diff
@@ -0,0 +1,447 @@
+    public static final String METHOD_REGISTER_FOR_NOTIFICATIONS = "RegisterForNotifications";
+    public static final String METHOD_UNREGISTER_FOR_NOTIFICATIONS = "UnregisterForNotifications";
+    private static final String DATABASE_NOT_INITIALIZED = "Database not initialized";
+    private static final String PROPERTY_SDV_ENABLED = "persist.sys.gm.sdv_enable";
```
Shouldn't be used in this project

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/UDiscoveryService.java:
```diff
@@ -0,0 +1,447 @@
+    private static final CharSequence NOTIFICATION_CHANNEL_NAME = "UDiscoveryService";
+    private static final int NOTIFICATION_ID = 1;
+    public static final UUri TOPIC_NODE_NOTIFICATION = buildCreateTopicUri();
+    protected static boolean VERBOSE = Log.isLoggable(LOG_TAG, Log.VERBOSE);
```
move closer to the TAG

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/UDiscoveryService.java:
```diff
@@ -0,0 +1,447 @@
+
+    public static @NonNull UStatus errorStatus(@NonNull String tag, @NonNull String method, @NonNull UStatus status,
+                                               Object... args) {
+        android.util.Log.e(tag, join(method, status, args));
```
use Formatter.status method

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/UDiscoveryService.java:
```diff
@@ -0,0 +1,447 @@
+        return builder.build();
+    }
+
+    private static void logStatus(@NonNull String message, @NonNull UStatus status) {
```
Please check uCore service project how logStatus was added using Formatter.status

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/UDiscoveryService.java:
```diff
@@ -0,0 +1,447 @@
+        return (CompletableFuture<Void>) mEULink.connect()
+                .thenCompose(status -> {
+                    logStatus("uLink connect", status);
+                    Log.i(LOG_TAG, join(Key.MESSAGE, "uLink.isConnected()", Key.CONNECTION, mEULink.isConnected()));
```
extra log, can be removed

jvnaiduwin replied:
Removed extra log

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/UDiscoveryService.java:
```diff
@@ -0,0 +1,447 @@
+                        registerMethod(METHOD_REGISTER_FOR_NOTIFICATIONS, this::executeRegisterNotification),
+                        registerMethod(METHOD_UNREGISTER_FOR_NOTIFICATIONS, this::executeUnregisterNotification))
+                .exceptionally(e -> {
+                    Log.e(LOG_TAG, join("registerAllMethods", toStatus(e)));
```
why not use logStatus()? (Here and similar places)

jvnaiduwin replied:
Done

mishap4 commented on up-discovery-android/service/src/main/java/org/eclipse/uprotocol/core/udiscovery/UDiscoveryService.java:
```diff
@@ -0,0 +1,447 @@
+
+    private void executeUpdateProperty(@NonNull UMessage requestEvent,
+                                       @NonNull CompletableFuture<UPayload> future) {
+        Log.d(LOG_TAG, join(Key.EVENT, "executeUpdateProperty"));
```
Not consistent, some handling methods are printing debug log like this others not

jvnaiduwin replied:
This some thing team has to decide. I Just integrate API eclipse proto

jvnaiduwin replied:
Sure, we can implement.

Commit:
Unit Test

Unit Test coverage in total approx 99%

Commit:
Clear testbase class

Code clean for unused variable

Commit:
Rename

-Change artifactory name org.eclipse.uprotocol.core.udiscovery.
-UDiscovery code cleanup.

Commit:
Changed gradle dependency to reflect new names of repos

Commit:
Review comment address

Unit test coverage.
Review comments address.
refactor code and rename variable.
remove unused class name, code and resources.
remove ultifi key name.

jvnaiduwin reviewed (commented):
Few of review comment not address, because not sure about the usage like permission, but most of review comment addressed.

Commit:
Contribution doc updated

Commit:
CAR permission removed

Commit:
Remove all the constants

Included from proto

METHOD_ADD_NODES.
METHOD_DELETE_NODES.
METHOD_FIND_NODES.
METHOD_FIND_NODE_PROPERTIES.
METHOD_LOOKUP_URI.
METHOD_REGISTER_FOR_NOTIFICATIONS.
METHOD_UNREGISTER_FOR_NOTIFICATIONS.
METHOD_UPDATE_NODE.
METHOD_UPDATE_PROPERTY.

Commit:
Review comment fix

logStatus added to utils.
Unit test coverage.
Log consistency in UDiscovery
Status Builder to buildStatus APi added

Commit:
DatabaseLoader class rename

We have two class one of which load Database.

Another load Database and AssetManagger we rename
it to ResourceLoader.

jvnaiduwin commented:
The latest up-java update has been incorporated. This fork happened on the latest release. Reason for closing this PR: If I include modifications, I must also include updates to up-core-android and up-example-android. 

Creating new PR.
